### ISSUE TITLE
Archive push issue after running out of disk

### ISSUE THREAD

**Issue by Anthares** [pgbackrest/pgbackrest#937]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
2.23 everywhere

2. PostgreSQL version:
11.6 on database A
11.2 on database B

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
Ubuntu 19.10 on database A
Debian 9.8 on database B
Debian 9.11 on repo

4. Did you install pgBackRest from source or from a package?
Always from package

5. Please attach the following as applicable:

#### `pgbackrest.conf` file(s)
```
# Repo configuration
[db-a-prod]
backup-standby=y
process-max=8
pg1-host=db-A
pg1-port=5432
pg1-host-user=postgres
pg1-path=/psql-data/11/main
pg2-host=db-A-replica
pg2-port=5432
pg2-host-user=postgres
pg2-path=/var/lib/postgresql/11/main

[db-B-prod]
process-max=4
pg1-host=db-B
pg1-port=5432
pg1-host-user=postgres
pg1-path=/var/lib/postgresql/11/main

[global]
repo1-cipher-pass=*redacted*
repo1-cipher-type=aes-256-cbc
repo1-path=/var/lib/pgbackrest
repo1-retention-full=1
repo1-retention-diff=1
start-fast=y

```
```
# DB A and replica
[db-A-prod]
pg1-path=/psql-data/11/main
recovery-option=primary_conninfo=host=db-A-replica port=5432 user=replicator

[global]
archive-async=y
log-level-file=detail
repo1-host=repo
spool-path=/var/spool/pgbackrest
archive-get-queue-max=1GB

[global:archive-get]
process-max=16

[global:archive-push]
process-max=8

```
```
# DB B
[db-B-prod]
pg1-path=/var/lib/postgresql/11/main

[global]
archive-async=y
log-level-file=detail
repo1-host=repo
spool-path=/var/spool/pgbackrest

[global:archive-get]
process-max=2

[global:archive-push]
process-max=2

```

#### `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
```
# DB A
# select name,setting from pg_settings where name in ('archive_command','archive_mode','wal_level','max_wal_senders');
      name       |                                  setting
-----------------+---------------------------------------------------------------------------
 archive_command | pgbackrest --stanza=db-A-prod --log-level-console=detail archive-push %p
 archive_mode    | on
 max_wal_senders | 10
 wal_level       | logical
(4 rows)

```

```
# DB B
# select name,setting from pg_settings where name in ('archive_command','archive_mode','wal_level','max_wal_senders');
      name       |                                  setting
-----------------+---------------------------------------------------------------------------
 archive_command | pgbackrest --stanza=db-B-prod --log-level-console=detail archive-push %p
 archive_mode    | on
 max_wal_senders | 10
 wal_level       | logical
(4 rows)

```
#### errors in the postgresql log file before or during the time you experienced the issue
On DB A :
```
ERROR: [082]: unable to push WAL file '00000002000024870000000A' to the archive asynchronously after 60 second(s)
2020-02-13 10:25:13.470 P00   INFO: archive-push command end: aborted with exception [082]
2020-02-13 10:25:13.472 UTC [14190] LOG:  archive command failed with exit code 82
2020-02-13 10:25:13.472 UTC [14190] DETAIL:  The failed archive command was: pgbackrest --stanza=db-A-prod --log-level-console=detail archive-push pg_wal/00000002000024870000000A
2020-02-13 10:25:14.484 P00   INFO: archive-push command begin 2.23: [pg_wal/00000002000024870000000A] --archive-async --log-level-console=detail --log-level-file=detail --pg1-path=/psql-data/11/main --process-max=8 --repo1-host=repo --spool-path=/var/spool/pgbackrest --stanza=db-A-prod

```
On DB B :
```
2020-02-13 11:29:56.292 P00   INFO: archive-push command begin 2.23: [pg_wal/00000001000010C1000000D7] --archive-async --log-level-console=detail --log-level-file=detail --pg1-path=/var/lib/postgresql/11/main --process-max=2 --repo1-host=repo --spool-path=/var/spool/pgbackrest --stanza=db-B-prod
ERROR: [099]: raised from local-1 protocol: raised from remote-1 protocol on 'repo': expected '{' at 'BRBLOCK168816'
2020-02-13 11:29:56.393 P00   INFO: archive-push command end: aborted with exception [099]
2020-02-13 11:29:56.396 CET [17291] LOG:  archive command failed with exit code 99
2020-02-13 11:29:56.396 CET [17291] DETAIL:  The failed archive command was: pgbackrest --stanza=db-B-prod --log-level-console=detail archive-push pg_wal/00000001000010C1000000D7

```
7. Describe the issue:

Hello,

Last night, our pgbackrest repo ran out of disk space.
I made some cleanup, by fixing ext4 default 5% reserved space (on 4.5 TB, this is huge) and running the expire command manually.
But the two databases using this repo are still unable to push wals using async command.

Do you have any idea in regards of the errors hereabove ? Logs on DB B make me think a file might be corrupted in the repo, but I have no idea which one and how to fix it

Thanks,
Best regards,

Nicolas

**Comment by dwsteele** [comment]
Do have any defunct pgbackrest processes on the pg hosts?  If so, you should manually kill them.

**Comment by Anthares** [comment]
Hello David,

I checked for and found idle pgbackrest processes on both databases.
Archive push resumed properly after killing them.

I launched diff backups on both databases. I hope they will succeed.
I'll update when they are done.

Thanks for the tips.
Regards,

Nicolas

**Comment by dwsteele** [comment]
Glad to hear it.  We believe we have fixed this issue at e2c304d473 which will be in the next release.

**Comment by Anthares** [comment]
Hello David,

Both backup went fine.
The linked commit might indeed fix this issue.

Thanks for your help.
Regards,

Nicolas
